controllers/storagecluster/consumer.go

```diff
@@ -0,0 +1,168 @@
+package storagecluster
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"google.golang.org/grpc"
+	"k8s.io/apimachinery/pkg/api/resource"
+	"k8s.io/apimachinery/pkg/types"
+
+	ocsv1 "github.com/red-hat-storage/ocs-operator/api/v1"
+	providerClient "github.com/red-hat-storage/ocs-operator/services/provider/client"
+	"github.com/red-hat-storage/ocs-operator/services/provider/common"
+)
+
+var (
+	// externalOCSResources will hold the ExternalResources for storageclusters
+	// ExternalResources can be accessible using the UID of an storagecluster
+	// getExternalConfigFromProvider func sets value for this once and use same again
+	externalOCSResources = map[types.UID][]ExternalResource{}
+)
+
+type ocsConsumer struct{}
+
+// ensureCreated performs onboarding and update capacity
+func (o *ocsConsumer) ensureCreated(r *StorageClusterReconciler, instance *ocsv1.StorageCluster) error {
```

Comment:
Let's separate it two function one that ensures the consumer is created and one that updates the capacity.

Reply:
Actually, everything will be called through the `ensureCreated` only as we are implementing the ` resourceManager` interface and we already have 2 different funcs for onboard and update capacity.

https://github.com/red-hat-storage/ocs-operator/blob/main/controllers/storagecluster/reconcile.go#L33